notification-server/src/bin/service.rs

```diff
@@ -112,28 +111,36 @@ async fn main() -> anyhow::Result<()> {
     let database_connection = DatabaseConnection::create(args.db_connection).await?;
 
     let mut concordium_client = Client::new(endpoint).await?;
-    let mut receiver = concordium_client.get_finalized_blocks().await?;
-    while let Some(v) = receiver.next().await {
-        let block_hash = v?.block_hash;
-        println!("Blockhash: {:?}", block_hash);
-        let transactions = concordium_client
-            .get_block_transaction_events(block_hash)
-            .await?
-            .response;
-        for result in process(transactions).await.iter() {
-            println!("address: {}, amount: {}", result.address, result.amount);
-            for device in database_connection
-                .prepared
-                .get_devices_from_account(result.address)
+    loop {
+        info!("Reading finalized blocks");
```

Comment:
This is important atm, since we want to detect time in which we may skip blocks.

When we persist state, this will no longer be relevant

Reply:
I guess the strategy here will be to cycle through a list of nodes similar to other services? I would suggest you look into using [TraverseConfig.new](https://docs.rs/concordium-rust-sdk/latest/concordium_rust_sdk/indexer/struct.TraverseConfig.html#method.new) along with [TransactionIndexer](https://docs.rs/concordium-rust-sdk/latest/concordium_rust_sdk/indexer/struct.TransactionIndexer.html) to get the results.

I completely forgot about this when we had our other discussion with regards to handling connection errors and actually features many of the mechanisms you wanted. Have a look at the documentation for [TraverseConfig.traverse](https://docs.rs/concordium-rust-sdk/latest/concordium_rust_sdk/indexer/struct.TraverseConfig.html#method.traverse), which is the method initiating the traversal.

Reply:
The idea on this line is to make an indication that we are now starting to read from the chain. 

The comment just addresses that this can be used for debugging given a notification has not occurred. 

We should not write in comment why we log. That sounds meta. 

We know we want to include persistent traversing, however this PR is well scoped, and fixes a critical issue, and changing that is not within such scope.

Reply:
That said using transaction indexer seems like a solid plan. But I would like to get this one merged first, as it literally causes issues. :)

Reply:
I am a bit twisted as to whether I should be using the loadbalancer or I should be using the traverser.